server/http_handler.go

```diff
@@ -268,6 +259,60 @@ func sessionHandler(w http.ResponseWriter, r *http.Request) {
 	jsonResult(w, res)
 }
 
+// deleteSessionHandler removes session in sessions table with given id
+func deleteSessionHandler(w http.ResponseWriter, r *http.Request) {
+	if dbserver.Instance == nil {
+		jsonError(w, http.StatusBadRequest, errors.New("database offline"))
+		return
+	}
+
+	var res db.Sessions
+
+	vars := mux.Vars(r)
+	id := vars["id"]
+
+	if txn := dbserver.Instance.Table("sessions").Delete(&res, id); txn.Error != nil {
+		jsonError(w, http.StatusBadRequest, txn.Error)
+		return
+	}
+
+	jsonResult(w, res)
+}
+
+// updateSessionHandler updates the data of an existing session
+func updateSessionHandler(w http.ResponseWriter, r *http.Request) {
+	if dbserver.Instance == nil {
+		jsonError(w, http.StatusBadRequest, errors.New("database offline"))
+		return
+	}
+
+	vars := mux.Vars(r)
+	id := vars["id"]
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		jsonError(w, http.StatusBadRequest, err)
+		return
+	}
+
+	var requestBody map[string]any
+
+	if err := json.Unmarshal(body, &requestBody); err != nil {
+		jsonError(w, http.StatusBadRequest, err)
+		return
+	}
+
+	sessionFields := map[string]interface{}{}
+	for key, value := range requestBody {
```

Comment:
Was ist der Unterschied zu requestBody?

Reply:
requestBody ist das interpretierte JSON und sessionFields beinhaltet alle Felder aus dem JSON und liefert das Format, das gorm für das Update der Zeile braucht

Reply:
{
  vehicle: "Tesla"
}

[
  "vehicle": "Tesla"
]

Reply:
Ich habe es gerade nochmal ausprobiert. Es geht auch ohne den extra Schritt 👍

Reply:
Weil beides den gleiichen typ hat- map of string

Reply:
Ja wobei das eine ja ne map of strings und die andere eine map of interface{} war

Reply:
Nein, genau nicht. Die sind absolut identisch

Reply:
Ah ok, wieder was gelernt 👍